components/org.wso2.carbon.siddhi.extensions.installer.core/src/main/java/org/wso2/carbon/siddhi/extensions/installer/core/config/mapping/models/ExtensionConfig.java

```diff
@@ -28,28 +29,30 @@
  */
 public class ExtensionConfig {
 
-    private Map<String, String> extension;
-    private List<DependencyConfig> dependencies;
+    private Map<String, String> extension = new HashMap<>();
+    private ExtensionIdentifierConfig identifier;
+    private List<DependencyConfig> dependencies = new ArrayList<>();
 
     public List<DependencyConfig> getManuallyInstallableDependencies() {
-        if (dependencies != null) {
-            return dependencies.stream().filter(dependency -> !dependency.isAutoDownloadable())
-                .collect(Collectors.toList());
-        }
-        return Collections.emptyList();
+        return dependencies.stream()
+            .filter(dependency -> !dependency.isAutoDownloadable())
```

Comment:
can't we make it to two lines?

Reply:
Updated in https://github.com/wso2/carbon-analytics/pull/1840/commits/687d9d726cb4be51cef81b489166d694c78b91dd